back/src/controller/user.controller.ts

```diff
@@ -14,4 +31,48 @@ const createUser = async (req: any, res: any) => {
   }
 }
 
+/**
+ * Login user
+ * @param req This is the request object
+ * @param res This is the response object
+ * @returns {Promise<void>} This returns the user object and a token if successful or an error message if unsuccessful
+ */
+const login = async (req: any, res: any): Promise<void> => {
+    try {
+        const {username, password} = req.body;
+        // check if username and password are provided
+        if (!username || !password) {
+            res.status(400).json({ error: "Please provide username and password" });
+            return;
+        }
+        const user = await User.findOne({
+            where: {
+                username: username,
+            }
+        });
+
+        // check if user exists
+        if (!user) {
+            res.status(401).json({ error: "User not found" });
+            return;
+        }
+
+        // check if password matches
+        const match: boolean = bcrypt.compareSync(password, user.password);
+        if (!match) {
+            res.status(401).json({ error: "Incorrect password" });
+            return;
+        }
+
+        // create token
+        const jwtSecret: string = process.env.JWT_SECRET || '';
+        const token: string = jwt.sign({ userId: user.id }, jwtSecret, { expiresIn: '1h' });
```

Comment:
yes but i already check before if the JWT_SECRET exist, and i can't directly assign process.env.JWT_SECRET beacause it could return undefined and i can't assign undefined to jswtSecret, how could I solve this issue ?